### DISCUSSION TITLE
Export ipynb to pdf

### DISCUSSION THREAD

**Discussion by ZahidUllah1992** [AammarTufail/python_ka_chilla2023#20] [Q&A]
I tried to export to pdf but the following errors, occur
Export failed. Please check the 'Jupyter' [output](command:jupyter.viewOutput) panel for further details.

Compared to anaconda, when i used jupyter notebook using chrome, but in VScode i tried various methods, but i am unable to convert it, 
I installed the tex, nbconvert, pandoc etc.
I also added the latex extension.
The command jupyter nbconvert --to pdf test.ipynb also did not work.
Please let me know
Though i have done using the jupyter notebook in anaconda

**Comment by AzamBukhari** [comment]
Almost all the Junyper extensions installed.

Cannot export to PDF from command line, Export option or command pallete (Ctrl+Shift+P) :(

**Reply by Hashirahmad996** [reply]
follow these steps to convert .ipynb to PDF directly [link](https://www.youtube.com/watch?v=EITY3U4lHus)

**Comment by BabarKhalidBaloch** [comment]
tried to export to pdf but the following errors, occur
Export failed. 
also try "MiKTeX" but did not work.

**Comment by Muhammad-Ali-Butt** [comment]
i have also tried the methods mentioned by @ZahidUllah1992 but the error still remains and can not directly export as pdf so kindly help.

**Reply by nouman073** [reply]
same error

**Reply by kaleemrao794** [reply]
Aslam o Alaikum

Kindly follow these instructions very carefully and you will found the solution
[test2.pdf](https://github.com/AammarTufail/python_ka_chilla2023/files/10359886/test2.pdf)

**Comment by baberibrar** [comment]
> I tried to export to pdf but the following errors, occur Export failed. Please check the 'Jupyter' output panel for further details.
> 
> Compared to anaconda, when i used jupyter notebook using chrome, but in VScode i tried various methods, but i am unable to convert it, I installed the tex, nbconvert, pandoc etc. I also added the latex extension. The command jupyter nbconvert --to pdf test.ipynb also did not work. Please let me know Though i have done using the jupyter notebook in anaconda

To export an IPython notebook (.ipynb file) to a PDF using mljar-mercury, you will need to install mljar-mercury and any other required dependencies. you can follow these steps:

- pip install mljar-mercury
- open the terminal where your .ipynb file save(directory) and run this cmd **mercury watch my_notebook.ipynb**
- Then your mercury run Please open [127.0.0.1:8000](http://127.0.0.1:8000/) to check the app running. Click the download button and download it into PDF. 
![Screenshot from 2023-01-06 16-48-32](https://user-images.githubusercontent.com/61284156/211006773-2ac72532-fe01-40ec-928e-6fcd5b9c5df4.png)
**source**: [https://github.com/mljar/mercury](url)

**Reply by BabarKhalidBaloch** [reply]
![aaaa](https://user-images.githubusercontent.com/110555681/211009769-c825f8b8-3397-4dcb-a09a-8c4097d97484.JPG)
after clicking download button..
any help ?

**Reply by Hashirahmad996** [reply]
follow these steps to convert .ipynb to PDF directly [link](https://www.youtube.com/watch?v=EITY3U4lHus)

**Reply by Muhammad-Ali-Butt** [reply]
@baberibrar i keep getting this error when i try to install mercury :
note: This error originates from a subprocess, and is likely not a problem with pip.       
  ERROR: Failed building wheel for gevent
Failed to build gevent
ERROR: Could not build wheels for gevent, which is required to install pyproject.toml-based projects

any help will be appreciated.

**Comment by SahilJatoi744** [comment]
Easiest way is to export .ipynb to PDF use online sources like:
https://www.vertopal.com/en/convert/ipynb-to-pdf

**Comment by usamabajwa86** [comment]
To convert an IPython Notebook (.ipynb) file to a PDF in Visual Studio Code, you will need to:

Install the Python extension for Visual Studio Code.
Install nbconvert by running !pip install nbconvert in a code cell.
Use the following code in a code cell:
Copy code
!jupyter nbconvert --to pdf "file_name.ipynb"
This will convert the .ipynb file to a PDF and save it in the same directory as the original file.

Alternatively, you can use the following command:

Copy code
!jupyter nbconvert --to latex --template latex "file_name.ipynb"
This will convert the .ipynb file to a LaTeX file, which you can then compile to a PDF using a LaTeX compiler such as pdflatex.

Note: If you don't want to use the command line, you can use a graphical tool such as [nbconvert GUI](https://github.com/tritemio/nbconvert-gui) to convert the .ipynb file to a PDF.

**Comment by Muhammad-Ali-Butt** [comment]
@ZahidUllah1992 @nouman073 @SahilJatoi744 @kaleemrao794 @AzamBukhari 

first download and install pandoc and miktex
then after miktex installation you will get a notification to update it **DO NOT UPDATE JUST IGNORE IT**
open vscode then on terminal write pip install TeX and after this write pip install nbconvert 
after this close vscode and open again then just simply follow this video : [click_here](https://www.youtube.com/watch?v=EITY3U4lHus)

**Comment by Hashirahmad996** [comment]
follow these steps to convert .ipynb to PDF directly [click here](https://www.youtube.com/watch?v=EITY3U4lHus)

**Reply by shahbaz-prog** [reply]
**To export ipynb file to pdf**
1- save the file in the visual code
2- run the Jupyter notebook from anaconda
3-on the top right we have **upload** option
4- upload the file.
5-scroll down to find the saved file to open it.
6-go to file and select print preview to export file to pdf.

**Reply by nouman073** [reply]
thanks

**Comment by AzamBukhari** [comment]
![Error_IPYNB_to_PDF](https://user-images.githubusercontent.com/121688289/211160201-0d9e1303-e40e-4182-b384-7725904eed19.png)
Hello
Thank you for sharing the link.

WHAT I HAVE ALREADY DONE:
-All Extensions & LIBRARIES installed successfully
-Installed Miktex, RESTARTED VS-Code & My laptop multiple times
-All Packages as per the youtube video downloaded succesfully
-While watching the youtube video at 6 minute, when the video instructor is saying a window will appear asking for packages
 * No window appeared (is anyone else having same issue)
 * I am using Windows11
 * I am getting the attached error

>

**Reply by usamabajwa86** [reply]
Have you installed nbconvert from nbconvert website?

**Reply by AzamBukhari** [reply]
pip install nbconvert

**Reply by Muhammad-Ali-Butt** [reply]
@AzamBukhari did you also install TeX ? if not then write pip install TeX and then follow the vid again

**Reply by AzamBukhari** [reply]
"Requirement already satisfied"
yaaaron kuch samajh nahin araha...
Sooie phus guie hai meri...
Aur ratein kaaali ho rahi hain...

Very frustrating

**Reply by AzamBukhari** [reply]
Can you guys share scrreen shot of your Windows PATH

**Reply by usamabajwa86** [reply]
also install pandoc

**Reply by Hashirahmad996** [reply]
@AzamBukhari  restart your pc then check again.i was facing same problem then after restart all set.

**Comment by AzamBukhari** [comment]
@ZahidUllah1992 @Hashirahmad996 @usamabajwa86 @Muhammad-Ali-Butt @SahilJatoi744 @kaleemrao794


yahāñ kotāhī-e-zauq-e-amal hai ḳhud giraftārī 
jahāñ baazū simaTte haiñ vahīñ sayyād hotā hai 


Bhaiyuon: Jazak Allah Khair


Thank you very much.

Finally accomplished the damn thing. With family & job it was not easy though.
But did it. And you guys will definitely laugh when I am going to share the ROOT CAUSE!!

C:\Users\azamb\OneDrive\Documents\PKC_2023\day_5> **jupyter-nbconvert --to pdf test.ipynb**

Just for future reference purpose I am attaching screenshot of my  Environment variables too



![Screenshot 2023-01-07 212420_environmentVariables](https://user-images.githubusercontent.com/121688289/211171131-64f51ed6-13a7-40a0-92e8-b5db885b8ee8.png)


Again many thanks
